app/javascript/flavours/glitch/features/notifications/index.js

```diff
@@ -224,8 +226,56 @@ class Notifications extends React.PureComponent {
     this.props.onMarkAsRead();
   };
 
+  /**
+   * Gets the list of notifications, grouped up (as per user settings) such that multiple users' interactions on the same
+   * post are collapsed into a single notification.
+   */
+  getGroupedNotifications() {
+    const { notifications, grouping } = this.props;
+    const groupedNotifications = [];
+
+    // if grouping is { "favourite": true, "reblog": false, "foo": true, "bar": false }
+    // then typesToGroup is [ "favourite", "foo" ]
+    const typesToGroup = grouping.reduce((acc, enabled, groupBy) => enabled ? acc.push(groupBy) : acc, ImmutableList.of());
+
+    // for each notification....
+    for (const notif of notifications) {
+
+      // `null` is used to signify that there is a "loading gap" in the notifications. We make sure that these loading gaps persist.
+      if (!notif) {
+        groupedNotifications.push(notif);
+        continue;
+      }
+
+      // Make sure that we only group up notifications of the provided types.
+      if (typesToGroup.includes(notif.get('type'))) {
+
+        // Get an already existing notification to collapse into
+        const matchingNotifIdx = groupedNotifications.findIndex(
+          other => other?.get('type') === notif.get('type') && other?.get('status') === notif.get('status'),
+        );
+        const matchingNotif = groupedNotifications[matchingNotifIdx];
+
+        // Collapse this notifcation into the existing notification if it exists,
+        // otherwise push it as a new notification.
+        if (matchingNotif) {
+          groupedNotifications[matchingNotifIdx] = matchingNotif.update(
+            'account',
+            ImmutableList(),
+            accounts => accounts.push(notif.get('account')),
+          );
+        } else {
+          groupedNotifications.push(notif.update('account', singleAccount => ImmutableList.of(singleAccount)));
+        }
+      } else {
+        groupedNotifications.push(notif);
+      }
+    }
+    return ImmutableList(groupedNotifications);
+  }
+
   render () {
-    const { intl, notifications, isLoading, isUnread, columnId, multiColumn, hasMore, numPending, showFilterBar, lastReadId, canMarkAsRead, needsNotificationPermission } = this.props;
+    const { intl, isLoading, isUnread, columnId, multiColumn, hasMore, numPending, showFilterBar, lastReadId, canMarkAsRead, needsNotificationPermission, grouping } = this.props;
```

Comment:
`grouping` is unused now.

```suggestion
    const { intl, isLoading, isUnread, columnId, multiColumn, hasMore, numPending, showFilterBar, lastReadId, canMarkAsRead, needsNotificationPermission } = this.props;
```